src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -2735,5 +2735,51 @@ public void PresentHeaders_BlocksDownloaded_ForkPresented_BlockDataForAlternativ
             Assert.Equal(chainedHeaderTree.UnconsumedBlocksDataBytes, serializedSizeOfChainsAandB);
         }
+
+        /// <summary>
+        /// Issue 55 @ CT is at 20a. AssumeValid is on 12b.
+        /// Peer 1 presents a chain with tip at 15b and fork point at 10a.
+        /// 11b,12b should be marked as AV, 13b,14b,15b should be header only.
+        /// No headers from chain b are requested for download.
+        /// </summary>
+        [Fact] public void PresentHeaders_AssumeValidBelowAVBlock_HeadersOnlyAboveAVBlock_NoHeadersRequestedForDownloaded()
+        {
+            const int initialChainSizeOfTwenty = 20;
+            const int chainExtensionSizeOfFive = 5;
+            const int assumeValidBlockHeightOfTwelve = 12;
+            
+            const int forkHeight = 10;
+            const int peerOneId = 1;
+
+            // Chain header tree setup.
+            TestContext testContext = new TestContextBuilder().WithInitialChain(initialChainSizeOfTwenty).UseCheckpoints(false).Build();
+            ChainedHeaderTree chainedHeaderTree = testContext.ChainedHeaderTree;
+            ChainedHeader initialChainTip = testContext.InitialChainTip;
+
+            // Chain tip is at 20.
+            Assert.Equal(initialChainSizeOfTwenty, initialChainTip.Height);
+
+            ChainedHeader tipOfChainBFork = testContext.ExtendAChain(chainExtensionSizeOfFive, initialChainTip.GetAncestor(forkHeight));
+
+            // AssumeValid is on 12b.
+            testContext.ConsensusSettings.BlockAssumedValid = tipOfChainBFork.GetAncestor(assumeValidBlockHeightOfTwelve).HashBlock;
+
+            // Peer 1 presents a chain with tip at 15b and fork point at 10a.
+            List<BlockHeader> listOfExtendedChainBlockHeaders = testContext.ChainedHeaderToList(tipOfChainBFork, tipOfChainBFork.Height - forkHeight + 1 /* fork inclusive */);
+            ConnectNewHeadersResult connectNewHeadersResult = chainedHeaderTree.ConnectNewHeaders(peerOneId, listOfExtendedChainBlockHeaders);
+
+            // 11b, 12b should be marked as AV.
+            connectNewHeadersResult.Consumed.GetAncestor(11).BlockValidationState.Should().Be(ValidationState.AssumedValid);
+            connectNewHeadersResult.Consumed.GetAncestor(12).BlockValidationState.Should().Be(ValidationState.AssumedValid);
+
+            // 13b, 14b, 15b should be header only.
+            connectNewHeadersResult.Consumed.GetAncestor(13).BlockValidationState.Should().Be(ValidationState.HeaderValidated);
+            connectNewHeadersResult.Consumed.GetAncestor(14).BlockValidationState.Should().Be(ValidationState.HeaderValidated);
+            connectNewHeadersResult.Consumed.GetAncestor(15).BlockValidationState.Should().Be(ValidationState.HeaderValidated);
+
+            // No headers from chain b are requested for download.
+            connectNewHeadersResult.DownloadFrom.Should().Be(null);
+            connectNewHeadersResult.DownloadTo.Should().Be(null);
+        }
```

Comment:
Very good, let's just again check block availability state for 11b – 15b

Reply:
Done.